src/operator/nn/pooling.cc

```diff
@@ -126,8 +126,8 @@ static bool PoolingShape(const nnvm::NodeAttrs &attrs,
       oshape[2] = 1 +
                   (dshape[2] + 2 * param.pad[0] - param.kernel[0]) /
                       param.stride[0];
-    } else {
-      oshape[2] = 1 + static_cast<int>(std::ceil(
+    } else if (param.pooling_convention == pool_enum::kFull) {
```

Comment:
You introduced "else if" and removed "else". So there is no default behavior here. Are you sure all the cases will fall under these 2 conditionals only ?